### PR TITLE
WIP: removes job scaler

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```
TBD

```

### CODE REVIEW

File: pkg/kubectl/delete.go [hunk 1]
```diff
@@ -82,9 +80,6 @@ func ReaperFor(kind schema.GroupKind, c internalclientset.Interface, sc scalecli
 	case api.Kind("Pod"):
 		return &PodReaper{c.Core()}, nil
 
-	case batch.Kind("Job"):
```
Comment (soltysh) [inline pkg/kubectl/delete.go]: I'm not quite sure about this being removed already. I never can't find the guarantees we give wrt to backwards compatibility. @deads2k that reminds me we should probably clean the reapers, we have GC in for quite a while, but cli still manually handles these, which is bad imo.

Comment (p0lyn0mial) [reply pkg/kubectl/delete.go]: @soltysh I have found the website that describes deprecation policy https://kubernetes.io/docs/reference/deprecation-policy/

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -2791,8 +2790,8 @@ func ScaleResource(
 	gr schema.GroupResource,
 ) error {
 	By(fmt.Sprintf("Scaling %v %s in namespace %s to %d", kind, name, ns, size))
-	scaler := kubectl.ScalerFor(kind, internalClientset.Batch(), scalesGetter, gr)
-	if err := testutils.ScaleResourceWithRetries(scaler, ns, name, size); err != nil {
+	scaler := kubectl.NewScaler(scalesGetter)
```
Comment (deads2k) [inline test/e2e/framework/util.go]: woohoo!

File: pkg/kubectl/delete.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/kubectl/delete.go]: removing code feels so good :)

File: test/e2e/framework/util.go (hunk 2 above)
Comment (p0lyn0mial) [reply test/e2e/framework/util.go]: yeah, that was a good idea :)

**Pull request closed without merging**